src/bp/core/services/ghost/memory-cache.ts

```diff
@@ -36,29 +35,37 @@ export default class MemoryObjectCache implements ObjectCache {
   }
 
   async set<T>(key: string, obj: T): Promise<void> {
-    this.cache.set(key, obj)
-    this.events.emit('invalidation', key)
+    const keyAlreadyExist = await this.has(key)
+
+    const keyUpdated = this.cache.set(key, obj)
+
+    if (keyAlreadyExist && keyUpdated) {
+      this.events.emit(EVENTS.invalidation, key)
+    }
   }
```

Comment:
I just checked LRU-cache, and all the methods are not async... we could remove a couple of async / await in this class